codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2LocalFlowController.java

```diff
@@ -195,14 +195,26 @@ private FlowState connectionState() {
     }
 
     private static FlowState state(Http2Stream stream) {
-        checkNotNull(stream, "stream");
-        return stream.getProperty(FlowState.class);
+        if (stream instanceof FastPropertyAccess) {
```

Comment:
> I'm not sure that defining interfaces for LocalFlowState and RemoteFlowState buy us anything, since none of the data is needed outside of the flow controllers. We might as well just store them as Object in the stream and provide template methods for access. WDYT?

This sounds reasonable.  I like having `LocalFlowState` and `RemoteFlowState` because it provides context as to what they are...and the casting will have to be done somewhere. This way the one that uses the object (and actually knows the type) can cast to what it knows...rather than allow/confuse others as to what is going on an try to cast to something else.

> worth seeing what it looks like.

You will see :)  The only not so nice thing is there is a bit more coupling between `Http2Connection` and the flow controllers for the `local()` and `remote()` endpoints.  I added an extra `listener` interface so the `Http2Connection` can be notified when the flow controllers are set for each endpoint...and this is used to create the `*FlowState` for the connection stream.  We can think on this a bit...but I'm not sure it is the worst thing because in "real" setups this is required to create new streams....